### PR TITLE
Use arrow keys to change focused comment in CommentThread overlay

### LINKED ISSUE #123: Use arrow key instead of tab to navigate inside a comment thread
Testing https://github.com/Microsoft/vscode/issues/55396 on Ubuntu

- Choose a comment which has replies
- Click on it in the comment panel. Focus moves to the comment in the editor
- At this point tab moves the focus to the reply

On tab, the focus usually is meant to move to the next actionable item. The comment and its replies are more like a list which should be navigate-able using arrow keys rather than tab

### PR DESCRIPTION
**Pull request by JacksonKearl**
Fixes Microsoft/vscode-pull-request-github#123. Allows navigating comment threads by using arrow keys, then tabbing to the action bar items. This is in line with the extensions viewlet's accessibility model.

### CODE REVIEW

File: src/vs/workbench/contrib/comments/browser/commentThreadWidget.ts [hunk 1]
```diff
@@ -631,8 +659,15 @@ export class ReviewZoneWidget extends ZoneWidget implements ICommentThreadWidget
 			if (this._commentThread.comments!.length === 0) {
 				this.dispose();
 			}
+
+			if (deletedElementIndex === this._focusedComment) { this.setFocusedComment(undefined); }
```
Comment (RMacfarlane) [inline src/vs/workbench/contrib/comments/browser/commentThreadWidget.ts]: It looks like `onDidDelete` is actually never trigged from the comment node :O this is left over from a previous version of the comment API where edit and delete were explicitly implemented by the extension, now the extension just provides a list of commands. Missed removing this, sorry!

The `update` method below should get called when comments are removed

**Review by RMacfarlane — CHANGES_REQUESTED** [review]
Nice! Just had a couple of comments

File: src/vs/workbench/contrib/comments/browser/commentThreadWidget.ts [hunk 2]
```diff
@@ -84,6 +86,17 @@ export class ReviewZoneWidget extends ZoneWidget implements ICommentThreadWidget
 	private _commentEditorIsEmpty!: IContextKey<boolean>;
 	private _commentFormActions!: CommentFormActions;
 	private _scopedInstatiationService: IInstantiationService;
+	private _focusedComment: number | undefined = undefined;
+	private setFocusedComment(value: number | undefined) {
+		if (this._focusedComment !== undefined) {
+			this._commentElements[this._focusedComment!].setFocus(false);
```
[the remark this answers is not in the document]
Comment (JacksonKearl) [reply src/vs/workbench/contrib/comments/browser/commentThreadWidget.ts]: Fixed

File: src/vs/workbench/contrib/comments/browser/commentThreadWidget.ts (hunk 1 above)
Comment (JacksonKearl) [reply src/vs/workbench/contrib/comments/browser/commentThreadWidget.ts]: I removed the `onDidDelete` stuff, and added more bounds checking to the setter.

**Review by RMacfarlane — APPROVED** [review]
LGTM!

File: src/vs/workbench/contrib/comments/browser/commentThreadWidget.ts [hunk 3]
```diff
@@ -401,6 +406,21 @@ export class ReviewZoneWidget extends ZoneWidget implements ICommentThreadWidget
 
 		this._commentsElement = dom.append(this._bodyElement, dom.$('div.comments-container'));
 		this._commentsElement.setAttribute('role', 'presentation');
+		this._commentsElement.tabIndex = 0;
+
+		dom.addDisposableListener(this._commentsElement, dom.EventType.KEY_DOWN, (e) => {
```
Comment (rebornix) [inline src/vs/workbench/contrib/comments/browser/commentThreadWidget.ts]: It might be added to `this._disposables`.

Comment (JacksonKearl) [reply src/vs/workbench/contrib/comments/browser/commentThreadWidget.ts]: @rebornix done, thanks!

**Pull request merged**